src/analysis/sensitivity.jl

```diff
@@ -434,8 +453,16 @@ function outcome_map(
 
     X_q = zeros(S + 1)
     for (j, fact_t) in enumerate(target_factors)
-        X_q .= quantile(X[:, fact_t], steps)
-        for (i, s) in enumerate(X_q[1:end-1])
+        ptype = model_spec.ptype[model_spec.fieldname.==fact_t][1]
+        if ptype == "categorical"
+            lb = foi_spec.lower_bound[foi_spec.fieldname.==fact_t][1]
+            ub = foi_spec.upper_bound[foi_spec.fieldname.==fact_t][1]
+            X_q .= round.(quantile(lb:1:ub, steps)) .-1
```

Comment:
We're re-checking against `foi_spec.fieldname.==fact_t` here.

Assign to a variable for readability

Reply:
```suggestion
        ptype = model_spec.ptype[model_spec.fieldname.==fact_t][1]
        if ptype == "categorical"
            lb = foi_spec.lower_bound[foi_spec.fieldname.==fact_t][1]
            ub = foi_spec.upper_bound[foi_spec.fieldname.==fact_t][1]
            X_q .= round.(quantile(lb:1:ub, steps)) .- 1
```

Missing space.